### PR TITLE
[MAIN-1792] Security update for holmes

### PR DESCRIPTION
**Pull request by Avi-Robusta**
We might want to do something different with the binaries for kube-lineage
tested in arm (locally) and amd
Notes:
- created a version of kube-lineage without go cve's
- Removed krew since it was no longer needed with our kubelineage binary
- removed gcloud and aws cli due to cves, we dont need them in cluster (I made a seperate docker image for local running)
- updated packages urllib3 and certifi due to cves

PR for kube lineage
https://github.com/tohjustin/kube-lineage/pull/14
my Kube lineage fork:
https://github.com/Avi-Robusta/kube-lineage

### CODE REVIEW

File: .github/workflows/build-docker-images.yaml [hunk 1]
```diff
@@ -37,6 +37,17 @@ jobs:
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v1
 
+    - name: Build and push Docker images
+      uses: docker/build-push-action@v2
+      with:
+        file: Dockerfile.local
```
Comment (arikalon1) [inline .github/workflows/build-docker-images.yaml]: can you rename to `Dockerfile.dev`?

File: .github/workflows/build-docker-images.yaml [hunk 2]
```diff
@@ -37,6 +37,17 @@ jobs:
...
+        context: .
+        platforms: linux/arm64,linux/amd64
+        push: true
+        tags: us-central1-docker.pkg.dev/genuine-flight-317411/devel/holmes_local:${{ github.ref_name }}
```
Comment (arikalon1) [inline .github/workflows/build-docker-images.yaml]: same, can you rename to `homes-dev`?

File: Dockerfile [hunk 3]
```diff
@@ -1,84 +1,105 @@
...
+ENV PATH="/root/.local/bin/:$PATH"
+
+RUN apt-get update \
+    && apt-get install -y \
+       curl \
+       git \
+       apt-transport-https \
+       gnupg2 \
+       build-essential \
+       unzip
```
Comment (arikalon1) [inline Dockerfile]: do we need to add:
```
  && apt-get purge -y --auto-remove \
  && rm -rf /var/lib/apt/lists/*
```
for cleanup?

File: Dockerfile [hunk 4]
```diff
@@ -1,84 +1,105 @@
...
 
+COPY --from=builder /app/venv /venv
 COPY . /app
 
+
+RUN apt-get update \
+    && apt-get install -y \
+       git \
+       apt-transport-https \
+       gnupg2
```
Comment (arikalon1) [inline Dockerfile]: do we need to add:
```
  && apt-get purge -y --auto-remove \
  && rm -rf /var/lib/apt/lists/*
```
for cleanup?

File: Dockerfile [hunk 5]
```diff
@@ -1,84 +1,105 @@
...
+
+# Set up kubectl
+COPY --from=builder /app/Release.key Release.key
+RUN cat Release.key |  gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
+    && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
+    && apt-get update
+RUN apt-get install -y kubectl
```
Comment (arikalon1) [inline Dockerfile:85]: any reason we're doing it here and not on the first `apt-get`?

File: Dockerfile [hunk 6]
```diff
@@ -1,84 +1,105 @@
...
+    && apt-get install -y \
+       git \
+       apt-transport-https \
+       gnupg2
+
+# Set up kubectl
+COPY --from=builder /app/Release.key Release.key
+RUN cat Release.key |  gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
+    && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
+    && apt-get update
```
Comment (arikalon1) [inline Dockerfile:84]: why do we need `apt-get update` at the end?

File: Dockerfile.local [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+ENV PATH="/root/.local/bin/:$PATH"
+
+RUN apt-get update \
+    && apt-get install -y \
+       curl \
+       git \
+       apt-transport-https \
+       gnupg2 \
+       build-essential \
+       unzip
```
Comment (arikalon1) [inline Dockerfile.local]: do we need to add:
```
  && apt-get purge -y --auto-remove \
  && rm -rf /var/lib/apt/lists/*
```
for cleanup?

File: Dockerfile.local [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+COPY --from=builder /app/venv /venv
+COPY . /app
+
+
+RUN apt-get update \
+    && apt-get install -y \
+       git \
+       apt-transport-https \
+       gnupg2
```
Comment (arikalon1) [inline Dockerfile.local]: do we need to add:
```
  && apt-get purge -y --auto-remove \
  && rm -rf /var/lib/apt/lists/*
```
for cleanup?

File: README.md [hunk 9]
```diff
@@ -157,10 +157,10 @@ holmes ask "what issues do I have in my cluster"
 <details>
 <summary>Prebuilt Docker Container</summary>
 
-Run the prebuilt Docker container `docker.pkg.dev/genuine-flight-317411/devel/holmes`, with extra flags to mount relevant config files (so that kubectl and other tools can access AWS/GCP resources using your local machine's credentials)
+Run the prebuilt Docker container `docker.pkg.dev/genuine-flight-317411/devel/holmes_local`, with extra flags to mount relevant config files (so that kubectl and other tools can access AWS/GCP resources using your local machine's credentials)
```
Comment (arikalon1) [inline README.md]: I would rename to dev

File: README.md [hunk 10]
```diff
@@ -157,10 +157,10 @@ holmes ask "what issues do I have in my cluster"
...
 
 ```bash
-docker run -it --net=host -v ~/.holmes:/root/.holmes -v ~/.aws:/root/.aws -v ~/.config/gcloud:/root/.config/gcloud -v $HOME/.kube/config:/root/.kube/config us-central1-docker.pkg.dev/genuine-flight-317411/devel/holmes ask "what pods are unhealthy and why?"
+docker run -it --net=host -v ~/.holmes:/root/.holmes -v ~/.aws:/root/.aws -v ~/.config/gcloud:/root/.config/gcloud -v $HOME/.kube/config:/root/.kube/config us-central1-docker.pkg.dev/genuine-flight-317411/devel/holmes_local ask "what pods are unhealthy and why?"
```
Comment (arikalon1) [inline README.md]: I would rename to dev

File: README.md [hunk 11]
```diff
@@ -184,7 +184,7 @@ Clone the project from github, and then run:
 
 ```bash
 cd holmesgpt
-docker build -t holmes .
+docker build -t holmes . -f Dockerfile.local
```
Comment (arikalon1) [inline README.md]: I would rename to dev

File: Dockerfile [hunk 12]
```diff
@@ -1,84 +1,105 @@
...
-# Add Krew to PATH
-ENV PATH="/root/.krew/bin:$PATH"
 
-# Install kube-lineage via Krew
-RUN kubectl krew install lineage
-
-# Copy the poetry configuration files into the container at /app
-COPY pyproject.toml poetry.lock* /app/
+# Set the architecture-specific kube lineage URLs
+ARG ARM_URL=https://github.com/Avi-Robusta/kube-lineage/releases/download/v2.0.1/kube-lineage-macos-latest-v2.0.1
```
Comment (arikalon1) [inline Dockerfile:30]: Do we still need the aws and gcp CLIs on the builder image?

**Review by arikalon1 — APPROVED** [review]
nice work

I think it works correctly, left some minor comments

File: Dockerfile (hunk 12 above)
Comment (Avi-Robusta) [reply Dockerfile:30]: no, ill remove it (but the above urls are kube-lineage)

File: Dockerfile (hunk 6 above)
Comment (Avi-Robusta) [reply Dockerfile:84]: we just added the kubernetes repo to apt 
`tee /etc/apt/sources.list.d/kubernetes.list`
so we needed update the package index to install kubectl

File: Dockerfile (hunk 5 above)
Comment (Avi-Robusta) [reply Dockerfile:85]: yea, added a comment, 
I am pretty sure that gnupg2 is gpg which is needed to install the kubernetes apt package but i can double check. 
so the flow is install gpg, add kubernetes apt registery + update packages than we can install kubectl
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#installing-kubeadm-kubelet-and-kubectl

**Pull request merged**